syntax = "proto3";
import "google/protobuf/timestamp.proto";

option go_package = "github.com/tensorland/modelbox/sdk-go/proto";

package modelbox;

// The RPC interface used by the workers
service ModelBoxAdmin {
  // Register an agent capable of running plugins
  rpc RegisterAgent(RegisterAgentRequest) returns (RegisterAgentResponse);
  // Workers heartbeat with the server about their presence
  // and work progress periodically
  rpc Heartbeat(HeartbeatRequest) returns (HeartbeatResponse);
}

message NodeInfo {
  string host_name = 1;
  string arch = 2;
}

message HeartbeatRequest {
  NodeInfo node = 1;
  google.protobuf.Timestamp at = 20;
}

message HeartbeatResponse {}

message SubscribeEventRequest {
  string namespace = 1;
  string ml_framework = 2;
  string owner = 3;
  repeated string actions = 4;
}

message RegisterAgentRequest {
}

message RegisterAgentResponse{

}